packages/hooks/src/useWalletMigration.ts

```diff
@@ -0,0 +1,43 @@
+import { useCallback } from 'react';
+import { useFetch, useInvalidateQuery, useRequest } from '@deriv/api';
+
+/** A custom hook to get the status of wallet_migration API and to start/reset the migration process */
+const useWalletMigration = () => {
+    const invalidate = useInvalidateQuery();
+
+    /** Make a request to wallet_migration API and onSuccess it will invalidate the cached data  */
+    const { mutate } = useRequest('wallet_migration', { onSuccess: () => invalidate('wallet_migration') });
+
+    /** Fetch the wallet_migration API and refetch it every second if the status is in_progress */
+    const { data } = useFetch('wallet_migration', {
+        payload: { wallet_migration: 'status' },
+        options: {
+            refetchInterval: response => (response?.wallet_migration?.status === 'in_progress' ? 500 : false),
```

Comment:
@thisyahlen-deriv Won't we get a rate limit? 500ms is so short in my opinion 🤔 

@yashim-deriv What do you think?

Reply:
@yashim-deriv said its fine 😄